packages/webapp/next.config.mjs

```diff
@@ -1,7 +1,28 @@
+const {
+  VERCEL_ENV = 'development',
+  VERCEL_URL = 'localhost'
+} = process.env
+
+let ROOT_URI
+
+switch (VERCEL_ENV) {
+  case "preview":
+    ROOT_URI = `https://${VERCEL_URL}/api/graphql`;
+    break;
+  case "production":
+    ROOT_URI = `https://gnars.wtf/api/graphql`;
+    break;
+  default:
+    ROOT_URI = `http://localhost:3000/api/graphql`;
```

Comment:
detect the port dynamically used from the `env` vars. If something is already running on port `3000` it will default to `3001`, etc.